pkg/compare/referenceV2.go

```diff
@@ -214,6 +217,85 @@ func (entry *FieldsToOmitV2Entry) process(previousKeys []string, toOmit *FieldsT
 	return paths, entry.processingError
 }
 
+type ReferenceTemplateV2 struct {
+	Config ReferenceTemplateConfigV2 `json:"config,omitempty"`
+	ReferenceTemplateV1
+}
+
+func (rf ReferenceTemplateV2) GetConfig() TemplateConfig {
+	return rf.Config
+}
+
+type ReferenceTemplateConfigV2 struct {
+	PerField []*PerFieldConfigV2 `json:"perField,omitempty"`
+	ReferenceTemplateConfigV1
+}
+
+func (config ReferenceTemplateConfigV2) GetInlineDiffFuncs() map[string]inlineDiffType {
+	diffFuncs := make(map[string]inlineDiffType)
+	for _, fieldConf := range config.PerField {
+		diffFuncs[fieldConf.PathToKey] = fieldConf.InlineDiffFunc
+	}
+	return diffFuncs
+}
+
+func (rf ReferenceTemplateV2) validateConfigPerField() error {
+	for pathToKey, inlineDiffFunc := range rf.GetConfig().GetInlineDiffFuncs() {
+		listedPath, err := pathToList(pathToKey)
+		if err != nil {
+			return fmt.Errorf("reference contains template with config per field with pathToKey that is not in "+
+				"supoorted format. path: %v", pathToKey)
+		}
+		value, exist, err := unstructured.NestedString(rf.metadata.Object, listedPath...)
+		if err != nil || !exist {
+			return fmt.Errorf("reference contains template with config per field with pathToKey that points to a "+
+				"path that does not exist in the template. path: %v", pathToKey)
+		}
+		validator, ok := InlineDiffFuncsValidators[inlineDiffFunc]
+		if !ok {
+			return fmt.Errorf("reference contains template with config per field with InlineDiffFunc that does not "+
+				"exist. InlineDiffFunc: %v", inlineDiffFunc)
+		}
+		if err := validator(value); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type PerFieldConfigV2 struct {
+	PathToKey      string         `json:"pathToKey,omitempty"`
+	InlineDiffFunc inlineDiffType `json:"inlineDiffFunc,omitempty"`
+}
+
+type inlineDiffType string
+
+const (
+	regex inlineDiffType = "regex"
+)
+
+var InlineDiffFuncs = map[inlineDiffType]func(string, string) string{regex: regexDiffFunc}
+var InlineDiffFuncsValidators = map[inlineDiffType]func(string) error{regex: regexDiffFuncValidator}
```

Comment:
I think these should actually be a single struct with func(probably diff or something) and validator as fields on it as they are required to come in pairs.